gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -121,13 +115,8 @@ protected static EntityResponse union(List<EntityResponse> entityResponses) {
             entityResponses.parallelStream()
                 .map(EntityResponse::getEntityFetcherResponse)
                 .collect(Collectors.toList()));
-    Set<EntityKey> entityKeys =
-        entityResponses.parallelStream()
-            .map(EntityResponse::getEntityKeys)
-            .reduce(Sets::union)
-            .orElse(Collections.emptySet());
 
-    return new EntityResponse(entityFetcherResponse, entityKeys);
+    return new EntityResponse(entityFetcherResponse, entityFetcherResponse.size());
```

Comment:
Can there be a case where the child DataFetcherNode set the total, but we are overriding the size again here?

Reply:
The union would happen when there is an OR node. 
The OR node would be there where there is an OR filter across different data sources. 
If there is an OR node across different data sources, the DataFetcherNode won't fetch total from the data store, since it only fetches total when pagination can be pushed down to the data store. 
The pagination will only be pushed down to the data store, when the filters are on the same data source

Reply:
Unless, the optimization for OR and AND for merging same data fetcher nodes goes for a toss

Reply:
I can be safe, and compute total, only if not set by the child data fetcher nodes, by using a `Long`, instead of `long`

Reply:
Isn't that safer?

Reply:
Yes. But, then if there are multiple data fetcher nodes, whose total to prefer?

Reply:
We can add an error LOG, if there are more than one data fetcher nodes for the same source, under OR and AND, so that we know that something is weird

Reply:
Ok

Reply:
ERROR log might be tricky. Let me add unit tests for OR and AND nodes

Reply:
added unit tests